curvefs/src/client/fuse_s3_client.cpp

```diff
@@ -230,9 +247,42 @@ CURVEFS_ERROR FuseS3Client::FuseOpMkNod(fuse_req_t req, fuse_ino_t parent,
     return MakeNode(req, parent, name, mode, FsFileType::TYPE_S3, e);
 }
 
+CURVEFS_ERROR FuseS3Client::FuseOpFsync(fuse_req_t req, fuse_ino_t ino,
+    int datasync, struct fuse_file_info *fi) {
+    LOG(INFO) << "fsync, ino = " << ino
+              << ", datasync = " << datasync;
+    std::shared_ptr<InodeWrapper> inodeWrapper;
+    CURVEFS_ERROR ret = inodeManager_->GetInode(ino, inodeWrapper);
+    if (ret != CURVEFS_ERROR::OK) {
+        LOG(ERROR) << "inodeManager get inode fail, ret = " << ret
+                  << ", inodeid = " << ino;
+        return ret;
+    }
+    ::curve::common::UniqueLock lgGuard = inodeWrapper->GetUniqueLock();
+    Inode inode = inodeWrapper->GetInodeUnlocked();
+    ret = s3Adaptor_->Flush(&inode);
+    if (ret != CURVEFS_ERROR::OK) {
+        LOG(ERROR) << "s3Adaptor_ flush failed, ret = " << ret
+                  << ", inodeid = " << inode.inodeid();
+        return ret;
+    }
+    inodeWrapper->SwapInode(&inode);
+    if (datasync != 0) {
+        return CURVEFS_ERROR::OK;
+    }
+    return inodeWrapper->Sync();
+}
+
 CURVEFS_ERROR FuseS3Client::Truncate(Inode *inode, uint64_t length) {
     return s3Adaptor_->Truncate(inode, 0);
 }
 
+void FuseS3Client::FlushData() {
+    CURVEFS_ERROR ret = CURVEFS_ERROR::UNKNOWN;
+    do {
+        ret = s3Adaptor_->FsSync();
```

Comment:
I think we need some protect strategy to prevent run `FsSync` forever.

Reply:
Break infinite loop here will lose data.